slimta/util/encoders.py

```diff
@@ -0,0 +1,52 @@
+from email.encoders import encode_base64 as _encode_base64
+
+import six
+
+__all__ = ['encode_base64']
+
+
+if six.PY2:
+    from email.encoders import _bencode
+
+    def fix_payload_decode(payload):
+        """ Inspired from pyton3.x email.message
+        """
+        try:
+            bpayload = payload.encode('ascii')
+        except UnicodeError:
+            # This won't happen for RFC compliant messages (messages
+            # containing only ASCII codepoints in the unicode input).
+            # If it does happen, turn the string into bytes in a way
+            # guaranteed not to fail.
+            bpayload = payload.encode('raw-unicode-escape')
+        return bpayload
+
+    def encode_base64(msg):
+        """Encode the message's payload in Base64.
+
+        Also, add an appropriate Content-Transfer-Encoding header.
+        """
+        orig = msg.get_payload(decode=True)
+        orig_ = orig[:]
+        # it is likely decode=True did nothing and orig is still unicode data
+        # (we expect bytes) acording to py2 implementation
+        if isinstance(orig, unicode):
+            orig = fix_payload_decode(orig)
+        encdata = unicode(_bencode(orig), 'ascii')
+        msg.set_payload(encdata)
+        msg['Content-Transfer-Encoding'] = 'base64'
+
+else:
+    def encode_base64(msg):
+        """ For some reason, the Py3 implementation of encode_base64 adds
+        "commodity newline". Let's remove it.
+        """
+        # mimicks py2 _bencode weirdness:
+        # if there is a \n on input, it is added on output
+        has_newline = (len(msg._payload) > 0) and (msg._payload[-1] == '\n')
+        _encode_base64(msg)
+
+        if not has_newline:
+            msg._payload = msg._payload[:-1]
```

Comment:
If this is just to make sure a test works, I would rather update the test to account for this newline. For huge message parts (e.g. video or image), this could be a fairly expensive operation.